### ISSUE TITLE
New build failures in build_modules 2.3.1

### ISSUE THREAD

**Issue by tvolkert** [dart-lang/build#2367]
The  version change  in `package:build_modues` from  `2.3.0` to `2.3.1` caused the following failure trying to build Flutter's  hello world example for the web:

```
$ flutter build web --release --no-pub --verbose
[  +20 ms] executing: [/Users/tvolkert/project/flutter/flutter/] git log -n 1 --pretty=format:%H
[  +29 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[        ] 82303af686422777fb717d969d5f3a4b647d2ae6
[        ] executing: [/Users/tvolkert/project/flutter/flutter/] git describe --match v*.*.* --first-parent --long --tags
[  +27 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.8.1-101-g82303af68
[  +10 ms] executing: [/Users/tvolkert/project/flutter/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +6 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] upstream/master
[        ] executing: [/Users/tvolkert/project/flutter/flutter/] git ls-remote --get-url upstream
[   +5 ms] Exit code 0 from: git ls-remote --get-url upstream
[        ] git@github.com:flutter/flutter.git
[  +38 ms] executing: [/Users/tvolkert/project/flutter/flutter/] git rev-parse --abbrev-ref HEAD
[   +5 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] fix_build
[  +93 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +5 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[  +55 ms] Compiling lib/main.dart for the Web...
[ +212 ms] Initializing inputs
[ +295 ms] Building new asset graph...
[ +467 ms] Building new asset graph completed, took 467ms

[        ] Checking for unexpected pre-existing outputs....
[   +3 ms] Checking for unexpected pre-existing outputs. completed, took 3ms

[  +12 ms] Running build...
[+1012 ms] 1.0s elapsed, 302/324 actions completed.
[+1086 ms] 2.1s elapsed, 687/693 actions completed.
[ +380 ms] Running dart2js with --libraries-spec="/Users/tvolkert/project/flutter/flutter/bin/cache/flutter_web_sdk/libraries.json" -O4 -o packages/hello_world/main_web_entrypoint.dart.js
--packages=".package-4928af1165300e4a579a862ef8487352" -Ddart.vm.product=true package:hello_world/main_web_entrypoint.dart

[   +6 ms] Running dart2js with --libraries-spec="/Users/tvolkert/project/flutter/flutter/bin/cache/flutter_web_sdk/libraries.json" -O4 -o packages/hello_world/arabic_web_entrypoint.dart.js
--packages=".package-afa4bbace903e1fd29402524afeff072" -Ddart.vm.product=true package:hello_world/arabic_web_entrypoint.dart

[+1013 ms] 3.5s elapsed, 865/867 actions completed.
[+1100 ms] 4.6s elapsed, 865/867 actions completed.
[+1103 ms] 5.7s elapsed, 865/867 actions completed.
[+1096 ms] 6.8s elapsed, 865/867 actions completed.
[+1101 ms] 7.9s elapsed, 865/867 actions completed.
[+1101 ms] 9.0s elapsed, 865/867 actions completed.
[+1100 ms] 10.1s elapsed, 865/867 actions completed.
[ +853 ms] Dart2Js finished with:

                    Compiled 14,834,695 characters Dart to 522,580 characters JavaScript in 8.31 seconds
                    Dart file package:hello_world/arabic_web_entrypoint.dart compiled to JavaScript: packages/hello_world/arabic_web_entrypoint.dart.js

[   +1 ms] Dart2Js finished with:

           Compiled 14,834,668 characters Dart to 522,487 characters JavaScript in 8.32 seconds
           Dart file package:hello_world/main_web_entrypoint.dart compiled to JavaScript: packages/hello_world/main_web_entrypoint.dart.js

[  +34 ms] building for target test/hello_test.dart.browser_test.dart
[ +443 ms] Error creating collection|lib/src/utils.ddc.dill
[   +6 ms] Error creating collection|lib/src/iterable_zip.ddc.dill
[   +4 ms] Error creating collection|lib/src/priority_queue.ddc.dill
[   +3 ms] Error creating collection|lib/src/algorithms.ddc.dill
[        ] Error creating meta|lib/meta.ddc.dill
[   +5 ms] Error creating collection|lib/src/comparators.ddc.dill
[        ] Error creating collection|lib/src/canonicalized_map.ddc.dill
[   +4 ms] Error creating collection|lib/collection.ddc.dill
[   +1 ms] Error creating flutter|lib/src/physics/clamped_simulation.ddc.dill
[   +3 ms] Error creating flutter|lib/physics.ddc.dill
[   +1 ms] Error creating vector_math|lib/hash.ddc.dill
[   +3 ms] Error creating vector_math|lib/vector_math_64.ddc.dill
[   +1 ms] Error creating typed_data|lib/typed_buffers.ddc.dill
[  +10 ms] Error creating path|lib/path.ddc.dill
[   +2 ms] Error creating pedantic|lib/pedantic.ddc.dill
[   +2 ms] Error creating charcode|lib/ascii.ddc.dill
[   +1 ms] Error creating charcode|lib/html_entity.ddc.dill
[  +11 ms] Error creating charcode|lib/charcode.ddc.dill
[   +1 ms] Error creating term_glyph|lib/src/generated/ascii_glyph_set.ddc.dill
[  +48 ms] Unable to read collection|lib/src/priority_queue.ddc.js, check your console or the `.dart_tool/build/generated/collection/lib/src/priority_queue.ddc.js.errors` log file.
[        ] Unable to read collection|lib/src/algorithms.ddc.js, check your console or the `.dart_tool/build/generated/collection/lib/src/algorithms.ddc.js.errors` log file.
[        ] Unable to read collection|lib/src/canonicalized_map.ddc.js, check your console or the `.dart_tool/build/generated/collection/lib/src/canonicalized_map.ddc.js.errors` log file.
[   +1 ms] Unable to read collection|lib/collection.ddc.js, check your console or the `.dart_tool/build/generated/collection/lib/collection.ddc.js.errors` log file.
[        ] Unable to read async|lib/async.ddc.js, check your console or the `.dart_tool/build/generated/async/lib/async.ddc.js.errors` log file.
[  +26 ms] Error creating flutter|lib/services.ddc.dill
[   +2 ms] Error creating flutter|lib/src/services/asset_bundle.ddc.dill
[   +2 ms] Error creating flutter|lib/foundation.ddc.dill
[   +1 ms] Error creating flutter|lib/src/foundation/_bitfield_web.ddc.dill
[  +10 ms] Error creating flutter|lib/src/animation/animation.ddc.dill
[  +15 ms] Error creating flutter|lib/animation.ddc.dill
[   +6 ms] Unable to read stack_trace|lib/src/chain.ddc.js, check your console or the `.dart_tool/build/generated/stack_trace/lib/src/chain.ddc.js.errors` log file.
[   +2 ms] Error creating flutter|lib/src/scheduler/binding.ddc.dill
[   +3 ms] Error creating async|lib/async.ddc.dill
[  +11 ms] Error creating flutter|lib/scheduler.ddc.dill
[   +4 ms] Error creating flutter|lib/src/gestures/arena.ddc.dill
[   +3 ms] Error creating flutter|lib/gestures.ddc.dill
[   +4 ms] Unable to read charcode|lib/html_entity.ddc.js, check your console or the `.dart_tool/build/generated/charcode/lib/html_entity.ddc.js.errors` log file.
[   +1 ms] Unable to read charcode|lib/charcode.ddc.js, check your console or the `.dart_tool/build/generated/charcode/lib/charcode.ddc.js.errors` log file.
[        ] Error creating flutter|lib/src/painting/_network_image_web.ddc.dill
[   +6 ms] Error creating flutter|lib/painting.ddc.dill
[   +8 ms] Error creating flutter|lib/src/semantics/binding.ddc.dill
[   +4 ms] Error creating flutter|lib/semantics.ddc.dill
[   +7 ms] Error creating stack_trace|lib/src/chain.ddc.dill
[  +22 ms] Unable to read source_span|lib/source_span.ddc.js, check your console or the `.dart_tool/build/generated/source_span/lib/source_span.ddc.js.errors` log file.
[        ] Error creating flutter|lib/src/rendering/animated_size.ddc.dill
[   +7 ms] Error creating flutter|lib/rendering.ddc.dill
[  +43 ms] Unable to read stream_channel|lib/stream_channel.ddc.js, check your console or the `.dart_tool/build/generated/stream_channel/lib/stream_channel.ddc.js.errors` log file.
[  +17 ms] Unable to read matcher|lib/src/core_matchers.ddc.js, check your console or the `.dart_tool/build/generated/matcher/lib/src/core_matchers.ddc.js.errors` log file.
[   +1 ms] Error creating stream_channel|lib/stream_channel.ddc.dill
[   +8 ms] Error creating matcher|lib/src/core_matchers.ddc.dill
[   +2 ms] Error creating matcher|lib/matcher.ddc.dill
[   +4 ms] Error creating quiver|lib/time.ddc.dill
[   +2 ms] Error creating quiver|lib/testing/async.ddc.dill
[   +6 ms] Error creating source_span|lib/source_span.ddc.dill
[   +2 ms] Error creating flutter|lib/widgets.ddc.dill
[   +7 ms] Error creating flutter|lib/src/widgets/actions.ddc.dill
[  +29 ms] Unable to read test_api|lib/src/suite_channel_manager.ddc.js, check your console or the `.dart_tool/build/generated/test_api/lib/src/suite_channel_manager.ddc.js.errors` log file.
[   +6 ms] Unable to read matcher|lib/matcher.ddc.js, check your console or the `.dart_tool/build/generated/matcher/lib/matcher.ddc.js.errors` log file.
[   +3 ms] Error creating test_api|lib/src/suite_channel_manager.ddc.dill
[   +6 ms] Unable to read string_scanner|lib/src/eager_span_scanner.ddc.js, check your console or the `.dart_tool/build/generated/string_scanner/lib/src/eager_span_scanner.ddc.js.errors` log file.
[   +9 ms] Error creating string_scanner|lib/src/eager_span_scanner.ddc.dill
[   +1 ms] Error creating hello_world|lib/main.ddc.dill
[   +6 ms] Unable to read vector_math|lib/vector_math_64.ddc.js, check your console or the `.dart_tool/build/generated/vector_math/lib/vector_math_64.ddc.js.errors` log file.
[   +4 ms] Error creating flutter|lib/src/cupertino/action_sheet.ddc.dill
[   +6 ms] Error creating flutter|lib/cupertino.ddc.dill
[  +15 ms] Unable to read flutter|lib/physics.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/physics.ddc.js.errors` log file.
[  +57 ms] Unable to read flutter|lib/src/foundation/_bitfield_web.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/foundation/_bitfield_web.ddc.js.errors` log file.
[  +24 ms] Error creating boolean_selector|lib/boolean_selector.ddc.dill
[  +25 ms] Unable to read flutter|lib/foundation.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/foundation.ddc.js.errors` log file.
[  +12 ms] Unable to read boolean_selector|lib/boolean_selector.ddc.js, check your console or the `.dart_tool/build/generated/boolean_selector/lib/boolean_selector.ddc.js.errors` log file.
[  +17 ms] Unable to read flutter|lib/src/services/asset_bundle.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/services/asset_bundle.ddc.js.errors` log file.
[  +25 ms] Unable to read flutter|lib/services.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/services.ddc.js.errors` log file.
[  +18 ms] Unable to read flutter|lib/src/scheduler/binding.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/scheduler/binding.ddc.js.errors` log file.
[   +5 ms] Unable to read test_api|lib/src/backend/metadata.ddc.js, check your console or the `.dart_tool/build/generated/test_api/lib/src/backend/metadata.ddc.js.errors` log file.
[   +4 ms] Error creating test_api|lib/src/backend/metadata.ddc.dill
[   +9 ms] Unable to read flutter|lib/scheduler.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/scheduler.ddc.js.errors` log file.
[  +16 ms] Unable to read flutter|lib/src/gestures/arena.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/gestures/arena.ddc.js.errors` log file.
[   +2 ms] Unable to read flutter|lib/gestures.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/gestures.ddc.js.errors` log file.
[  +11 ms] Unable to read flutter|lib/src/painting/_network_image_web.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/painting/_network_image_web.ddc.js.errors` log file.
[  +33 ms] Error creating flutter|lib/material.ddc.dill
[  +12 ms] Unable to read flutter|lib/painting.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/painting.ddc.js.errors` log file.
[   +3 ms] Unable to read test_api|lib/src/backend/closed_exception.ddc.js, check your console or the `.dart_tool/build/generated/test_api/lib/src/backend/closed_exception.ddc.js.errors` log file.
[   +6 ms] Unable to read flutter|lib/src/semantics/binding.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/semantics/binding.ddc.js.errors` log file.
[   +8 ms] Unable to read flutter|lib/semantics.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/semantics.ddc.js.errors` log file.
[  +14 ms] Unable to read flutter|lib/src/animation/animation.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/animation/animation.ddc.js.errors` log file.
[   +3 ms] Unable to read flutter|lib/animation.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/animation.ddc.js.errors` log file.
[   +8 ms] Unable to read flutter|lib/src/rendering/animated_size.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/rendering/animated_size.ddc.js.errors` log file.
[   +5 ms] Unable to read flutter|lib/rendering.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/rendering.ddc.js.errors` log file.
[  +12 ms] Unable to read flutter|lib/src/widgets/actions.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/widgets/actions.ddc.js.errors` log file.
[   +1 ms] Unable to read flutter|lib/widgets.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/widgets.ddc.js.errors` log file.
[   +1 ms] Unable to read hello_world|lib/main.ddc.js, check your console or the `.dart_tool/build/generated/hello_world/lib/main.ddc.js.errors` log file.
[  +22 ms] Unable to read quiver|lib/time.ddc.js, check your console or the `.dart_tool/build/generated/quiver/lib/time.ddc.js.errors` log file.
[  +22 ms] Unable to read quiver|lib/testing/async.ddc.js, check your console or the `.dart_tool/build/generated/quiver/lib/testing/async.ddc.js.errors` log file.
[        ] Unable to read flutter|lib/src/cupertino/action_sheet.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/src/cupertino/action_sheet.ddc.js.errors` log file.
[   +2 ms] Unable to read flutter|lib/cupertino.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/cupertino.ddc.js.errors` log file.
[        ] Unable to read flutter|lib/material.ddc.js, check your console or the `.dart_tool/build/generated/flutter/lib/material.ddc.js.errors` log file.
[+1013 ms] 13.3s elapsed, 1018/1023 actions completed.
[+1099 ms] 14.4s elapsed, 1018/1023 actions completed.
[+1098 ms] 15.5s elapsed, 1018/1023 actions completed.
```

**Comment by natebosch** [comment]
I'm not sure why we aren't seeing the error output from the kernel worker.

My best guess is https://github.com/dart-lang/build/pull/2306

cc @jonahwilliams - you use a different Platform than the DDC one right? If it's got a different name it probably doesn't match any target that the kernel worker knows about. We might need to make it possible to configure that differently.

**Comment by jonahwilliams** [comment]
Yes, I've configured a flutter web specific platform here:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_runner/build_script.dart#L42

**Comment by natebosch** [comment]
Ok, we need to determine a different way to configure the target name we use for kernel.

See original discussion at https://github.com/dart-lang/build/issues/2174

**Comment by natebosch** [comment]
Published as `2.4.0` with an option that will let you configure the target name - pass `'ddc'` for that argument when you construct the `kernelBuilder` and things should work again.
